Core/src/Surfaces/Surface.cpp

```diff
@@ -254,14 +254,10 @@ Acts::RotationMatrix3 Acts::Surface::referenceFrame(
 }
 
 Acts::BoundToFreeMatrix Acts::Surface::boundToFreeJacobian(
-    const GeometryContext& gctx, const BoundVector& boundParams) const {
-  // Transform from bound to free parameters
-  FreeVector freeParams =
-      detail::transformBoundToFreeParameters(*this, gctx, boundParams);
-  // The global position
-  const Vector3 position = freeParams.segment<3>(eFreePos0);
+    const GeometryContext& gctx, const FreeVector& parameters) const {
```

Comment:
We could consider naming this `freeParams`
```suggestion
    const GeometryContext& gctx, const FreeVector& freeParams) const {
```

but maybe not needed.

Reply:
I suppose one additional point is that the free-to-bound and bound-to-free roundtrip implicitly checks if we're on-surface. Should that still happen, or do we not care?

Reply:
I guess this is a question of the contract. I would think the free params are obtained by some bound params at some point so they should be bound. To me it sounds fair to keep this on the callers responsibility that the input is well defined because this is potentially hot code and we might not want to pay the price

Reply:
We could add an `assert` that only checks in debug?

Reply:
checked the repo and `const FreeVector& parameters` seems to be consistently used in the surface code right now

Reply:
I implemented this now - one potential problem with this is that the user might use a different tolerance and then we compare against our globally set tolerance. but maybe our tolerance is the upper bound for such things?